src/main/java/com/github/charithe/kafka/KafkaJunitRule.java

```diff
@@ -156,6 +175,64 @@ public ConsumerConfig consumerConfig() {
         props.put("auto.offset.reset", "smallest");
         return new ConsumerConfig(props);
     }
+    
+    /**
+     * Read messages from a given kafka topic as {@link String}.
+     * @param topic name of the topic
+     * @param expectedMessages number of messages to be read
+     * @return list of string messages
+     * @throws TimeoutException if no messages are read after 5 seconds
+     */
+    public List<String> readStringMessages(final String topic, final int expectedMessages) throws TimeoutException {
+        return readMessages(topic, expectedMessages, new StringDecoder(null));
+    }
+    
+    /**
+     * Read messages from a given kafka topic.
+     * @param topic name of the topic
+     * @param expectedMessages number of messages to be read
+     * @param decoder message decoder
+     * @return list of decoded messages
+     * @throws TimeoutException if no messages are read after 5 seconds
+     */
+    public <T> List<T> readMessages(final String topic, final int expectedMessages, final Decoder<T> decoder) throws TimeoutException {
+        ExecutorService singleThread = Executors.newSingleThreadExecutor();
+        ConsumerConnector connector = null;
+        try {
+            connector = Consumer.createJavaConsumerConnector(consumerConfig());
+
+            Map<String, List<KafkaStream<byte[], T>>> streams = connector.createMessageStreams(
+                    singletonMap(topic, 1), new DefaultDecoder(null), decoder);
+            
+            final KafkaStream<byte[], T> messageSteam = streams.get(topic).get(0);
+            
+            Future<List<T>> future = singleThread.submit(new Callable<List<T>>() {
+                @Override
+                public List<T> call() throws Exception {
+                    List<T> messages = new ArrayList<>(expectedMessages);
+                    ConsumerIterator<byte[], T> iterator = messageSteam.iterator();
+                    while (messages.size() != expectedMessages && iterator.hasNext()) {
+                        T message = iterator.next().message();
+                        LOGGER.info("Received message: {}", message);
```

Comment:
Would it be better to log this at `debug` level to avoid cluttering up the test output?